src/core/base-plugin.ts

```diff
@@ -75,7 +75,26 @@ export class BasePlugin {
     htmlFileName: string
     style: string
   }) {
-    const styleString = `<style type="text/css">${style}</style>`
+    let attributesString = ''
+
+    if (this.config.attributes) {
+      if (
+        typeof this.config.attributes !== 'object' &&
+        this.config.attributes === null
+      ) {
+        throw new Error(
+          `Please provide a key/value object if intending to use the attributes option`,
+        )
+      }
+
+      Object.keys(this.config.attributes).map((key) => {
+        const value = this.config.attributes[key] || ''
+        attributesString += ` ${key}="${value}"`
+      })
```

Comment:
`map` is not a good way to do `side effect` things, use `forEach` instead.

Reply:
A better way to get `attributesString` is extract a function to do this. Then we could avoid creating a temp variable.

```ts
function getAttributesString(config: Config): string {
  if (config.attributes && typeof config.attributes === 'object') {
    return Object
      .keys(config.attributes)
      .map((key) => `${key}="${config.attributes[key]}"`)
      .join(' ')
 }

  if (config.attributes !== undefined && config.attributes !== null) {
    throw new Error(
      `Please provide a key/value object if intending to use the attributes option, not ${config.attributes}`,
    )
  }
 
  return ''
}

protected addStyle() {
  ...
  const styleString = `<style ${getAttributesString(this.config)}>${style}</style>`
  ...
}
```